src/kafka/producer/metadata_manager.cc

```diff
@@ -41,13 +41,17 @@ namespace kafka {
     }
 
     seastar::future<> metadata_manager::refresh_coroutine(std::chrono::milliseconds dur) {
-        return seastar::async({}, [this, dur]{
-            while(_keep_refreshing) {
+        return seastar::do_until([this]{return ! _keep_refreshing;}, [this, dur]{
+            return seastar::sleep_abortable(dur, _stop_refresh).then([this] {
+                return refresh_metadata();
+            }).handle_exception([this] (std::exception_ptr ep) {
                 try {
-                    seastar::sleep_abortable(dur, _stop_refresh).get();
-                } catch (seastar::sleep_aborted& e) {}
-                refresh_metadata().wait();
-            }
+                    std::rethrow_exception(ep);
+                } catch (seastar::sleep_aborted& e) {
+                    return make_ready_future();
+                }
```

Comment:
Is it possible to get any other exception here?

Reply:
No, it is not as far as i know.

Reply:
If so then I'd advise removing the try/catch block completely and just returning `make_ready_future()`